src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -1225,7 +1223,7 @@ public FormValidation doCheckIncludes(@QueryParameter String value) {
         }
 
         @Restricted(NoExternalUse.class)
-        public FormValidation doCheckScanCredentialsId(@AncestorInPath SCMSourceOwner context,
+        public FormValidation doCheckScanCredentialsId(@CheckForNull @AncestorInPath Item context,
```

Comment:
Switching to `Item` should ensure that form validation & completion will work even on a library scoped to a plain folder when the credentials are also folder-scoped.